src/pynwb/spec.py

```diff
@@ -128,34 +128,44 @@ def __init__(self, **kwargs):
 
 
 class NWBDatasetSpec(BaseStorageOverride, DatasetSpec):
-    ''' The Spec class to use for NWB specifications '''
+    ''' The Spec class to use for NWB dataset specifications.
 
-    @docval(*_dataset_docval)
+    Classes will automatically include NWBData if None is specified.
+    '''
+
+    @docval(*deepcopy(_dataset_docval))
     def __init__(self, **kwargs):
-        args, kwargs = self._translate_kwargs(kwargs)
-        super(NWBDatasetSpec, self).__init__(*args, **kwargs)
+        kwargs = self._translate_kwargs(kwargs)
+        # set data_type_inc to NWBData only if it is not specified and the type is not an HDMF base type
+        if kwargs['data_type_inc'] is None and kwargs['data_type_def'] not in (None, 'Data'):
+            kwargs['data_type_inc'] = 'NWBData'
+        super(NWBDatasetSpec, self).__init__(**kwargs)
 
 
 _group_docval = __swap_inc_def(GroupSpec)
 
 
 class NWBGroupSpec(BaseStorageOverride, GroupSpec):
-    ''' The Spec class to use for NWB specifications '''
+    ''' The Spec class to use for NWB group specifications.
 
-    @docval(*_group_docval)
+    Classes will automatically include NWBContainer if None is specified.
+    '''
+
+    @docval(*deepcopy(_group_docval))
     def __init__(self, **kwargs):
-        args, kwargs = self._translate_kwargs(kwargs)
-        super(NWBGroupSpec, self).__init__(*args, **kwargs)
+        kwargs = self._translate_kwargs(kwargs)
+        # set data_type_inc to NWBData only if it is not specified and the type is not an HDMF base type
+        if kwargs['data_type_inc'] is None and kwargs['data_type_def'] not in (None, 'Container', 'CSRMatrix'):
```

Comment:
why is there a special case for `CSRMatrix` and not other types from hdmf-common?

Reply:
`Container`, `Data`, and `CSRMatrix` are the only types in hdmf-common without a `data_type_inc`. This will be fixed in hdmf-common-schema 1.2.1 from the just merged PR https://github.com/hdmf-dev/hdmf-common-schema/pull/40 , so it might be possible to remove `CSRMatrix` from the list here.

The if statement is necessary to prevent an infinite loop that occurs when loading the hdmf-common schema: Without this, loading `data_type_def: Container` with `data_type_inc: None` will result in `data_type_inc: NWBContainer`.

Reply:
Gotcha. Thanks for the clarifications